src/LandIce/evaluators/LandIce_UpdateZCoordinate_Def.hpp

```diff
@@ -201,4 +215,112 @@ evaluateFields(typename Traits::EvalData workset)
   }
 }
 
+//****************************************************************************
+
+template<typename EvalT, typename Traits>
+UpdateZCoordinateGivenTopAndBedSurfaces<EvalT, Traits>::
+UpdateZCoordinateGivenTopAndBedSurfaces (const Teuchos::ParameterList& p,
+                            const Teuchos::RCP<Albany::Layouts>& dl) :
+  coordVecIn (p.get<std::string> ("Old Coords Name"), dl->vertices_vector),
+  bedTopo(p.get<std::string> ("Bed Topography Name"), dl->node_scalar),
+  topSurf(p.get<std::string>("Top Surface Name"), dl->node_scalar),
+  coordVecOut(p.get<std::string> ("New Coords Name"), dl->vertices_vector),
+  H(p.get<std::string> ("Thickness Name"), dl->node_scalar)
+{
+  isTopSurfParam = p.isParameter("Top Surface Parameter Name");
+  isBedTopoParam = p.isParameter("Bed Topography Parameter Name");
+
+  if(isBedTopoParam == true) {
+    bedTopoIn = decltype(bedTopoIn)(p.get<std::string> ("Bed Topography Parameter Name"), dl->node_scalar);
+    this->addDependentField(bedTopoIn);
+    this->addEvaluatedField(bedTopo);
+  } else {
+    this->addDependentField(bedTopo);
```

Comment:
I'm curious: does this give a compiler error, since the dependent field is nor marked as const?

Reply:
Good question. No, I don't get a compiler error... we'll see if clang complains.

Reply:
No need. I just recalled PHX giving a warning (I think) if the data type of a dependent field was not marked as const. Maybe it was a deprecated method, I don't recall. But even if you did get it, fixing it would make the code a tad complicated (you'd have to use switches later on, to get a ref to bedTopo vs bedTopoIn...).

This can be left as is.